docs/tutorials/hpo-workflows/hpo-basic-integration.rst

```diff
@@ -0,0 +1,213 @@
+
+    from itwinai.parser import ConfigParser
+
+
+Step 2: Define the Trial Execution Function
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``run_trial`` function executes a single HPO trial using the provided configuration and data.
+``config`` contains the hyperparameters (e.g., learning rate, batch size). ``data`` is a general 
+dictionary that can contain any values you need to pass to the function. In our template, 
+it includes only the pipeline name defined in your config.yaml, but you may pass additional 
+values here. 
+
+**Note:** Keep the size of the data dictionary small because it is serialized by Ray Tune 
+when sent to workers. Avoid including large objects like datasets here. 
+
+.. code-block:: python
+
+    def run_trial(config: Dict, data: Dict):
+        pipeline_name = data["pipeline_name"]
+        parser = ConfigParser(
+            config="config.yaml",
+            override_keys={
+                # Set hyperparameters controlled by ray
+                "batch_size": config["batch_size"],
+                "optim_lr": config["optim_lr"]
+            },
+        )
+        my_pipeline = parser.parse_pipeline(
+            pipeline_nested_key=pipeline_name, verbose=False
+        )
+
+        my_pipeline.execute()
+
+**Adaptation**: 
+
+*    If neccessary, replace ``config.yaml`` with the path to your own configuration file.
+*    Adjust hyperparameter keys (e.g., ``batch_size``, ``learning_rate``) to match the keys used in your pipeline.
+*    If your pipeline requires additional configurations, add them to the ``override_keys`` dictionary.
+
+For a detailed explanation of defining and parsing pipelines, refer to the :doc:`Workflow Page <../../how-it-works/workflows/explain_workflows>`.
+
+
+.. _Step 3:
+
+Step 3: Configure and Run HPO
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The ``run_hpo`` function is the core of this script. It sets up Ray Tune to manage the hyperparameter optimization process.
+
+**1.  Define the search space:** 
+Specify the range of hyperparameters to explore. In our example:
+
+.. code-block:: python
+
+    search_space = {
+            "batch_size": tune.choice([3, 4, 5, 6]),
+            "optim_lr": tune.uniform(1e-5, 1e-3),
+        }
+
+**Adaptation**: 
+
+*    Add or remove hyperparameters to suit your model.
+*    Use ``tune.choice`` for discrete parameters and ``tune.uniform`` for continuous, uniform ranges. You can add different sampling distributions and types. For additional search space options, please refer to the `Ray Tune documentation <https://docs.ray.io/en/latest/tune/api/search_space.html>`_. Any search space definition supported by Ray Tune can be used in this template as well.
+
+**2.  Define ray configs:** 
+Set up the tuning and run configurations to control the number of trials, optimization goal, and stopping criteria.
+
+.. code-block:: python
+
+    tune_config = tune.TuneConfig(
+                metric=args.metric,  # Metric to optimize (loss by default)
```

Comment:
You don't necessarily need to change anything here, but was just wondering: What is the type of `args.metric` here? Is it a callable or something?

Reply:
No it is just a string ("min" or "max"). In the full code you can see this in the command-line arguments 😌